packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -44,23 +42,34 @@ class DirectStripeRoutes {
    * @param {*} push
    * @param {*} mailer
    * @param {*} profile
-   * @param {import('../payments/stripe')} payments
+   * @param {import('../payments/stripe')} stripeHelper
    */
-  constructor(log, db, config, customs, push, mailer, profile, payments) {
+  constructor(log, db, config, customs, push, mailer, profile, stripeHelper) {
     this.log = log;
     this.db = db;
     this.config = config;
     this.customs = customs;
     this.push = push;
     this.mailer = mailer;
     this.profile = profile;
-    this.payments = payments;
+    this.stripeHelper = stripeHelper;
 
     this.CLIENT_CAPABILITIES = Object.entries(
       config.subscriptions.clientCapabilities
     ).map(([clientId, capabilities]) => ({ clientId, capabilities }));
   }
 
+  async customerChanged(request, uid, email) {
+    await this.stripeHelper.deleteCachedCustomer(uid, email);
+    await this.profile.deleteCache(uid);
+    const devices = await request.app.devices;
```

Comment:
Moved this into a Promise.all - I *think* it should work in parallel with the other two, since they don't have any side-effects or sequence dependencies on each other